src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -1064,8 +1082,23 @@ AlgorithmPaintToolBox::GenerateMinMaxValuesFromImage ()
 
     m_wandLowerThresholdSlider->setRange(m_MinValueImage, m_MaxValueImage);
     m_wandUpperThresholdSlider->setRange(m_MinValueImage, m_MaxValueImage);
-    m_wandLowerThresholdSlider->setValue(m_MinValueImage);
-    m_wandUpperThresholdSlider->setValue(m_MinValueImage);
+    // Try to keep the previously selected range: current min
+    // and max values are kept if they both lie in the new min
+    // and max.
+    double currentMinValue = m_wandLowerThresholdSlider->value();
+    double currentMaxValue = m_wandUpperThresholdSlider->value();
+    if ((currentMinValue >= m_MinValueImage && currentMinValue < m_MaxValueImage) &&
+        (currentMaxValue >= m_MinValueImage && currentMaxValue < m_MaxValueImage))
+    {
+        m_wandLowerThresholdSlider->setValue(currentMinValue);
```

Comment:
so, no need to put theses values again? Or you want to trigger again the linked functions?